### PR TITLE
Fix command line argument handling for slice configuration parameters

### PR DESCRIPTION
**Pull request by the**
It seems slice configuration parameters such as `ExcludeDir` cannot be set correctly with command line arguments (nor override default config). The test for `--build.exclude_regex` only passes because the test value corresponds with the default value (set in `defaultConfig`), it's not actually set by the parameter in the test.

I propose to treat the command line arguments for such config parameters as comma separated strings. Setting a value would work as e.g. `--build.exclude_dir "templates,build"`.

### CODE REVIEW

**Comment by xiantang** [comment]
thx let me take a look

**Comment by xiantang** [comment]
ci not pass

**Comment by xiantang** [comment]
@the use comma to split string as slice is a good idea! plz fix the lint, and add the test case which include empty string, and seems should add more description in README.

after that, I will merge this PR. thanks

**Comment by the** [comment]
> ci not pass

I'm puzzled about the errors: `main.go:6:2: undefined: Println`. I haven't even changed that file and I don't see anything wrong with the `fmt.Println`'s in main 🤔. Do you have any idea?

File: runner/flag_test.go [hunk 1]
```diff
@@ -103,9 +103,9 @@ func TestConfigRuntimeArgs(t *testing.T) {
 		},
 		{
 			name: "check exclude_regex",
```
[the remark this answers is not in the document]
Comment (the) [reply runner/flag_test.go:105]: Hmm what would you expect in that case, the default values ore empty slice? It seems currently it keeps the default values (I didn't think of this case before), but I'd rather expect to set it to empty.

[the remark this answers is not in the document]
Comment (xiantang) [reply runner/flag_test.go:105]: I prefer set it to empty

**Comment by xiantang** [comment]
> > ci not pass
> 
> I'm puzzled about the errors: `main.go:6:2: undefined: Println`. I haven't even changed that file and I don't see anything wrong with the `fmt.Println`'s in main 🤔. Do you have any idea?

I have rerunned the failed test, It's passed. the failed pipeline seems not relate with this version. ignore it.

File: runner/flag_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (the) [reply runner/flag_test.go:105]: Unfortunately I think it isn't that straightforward. Before a value is set there's a check for nil and empty string https://github.com/cosmtrek/air/blob/master/runner/config.go#L324. If I only keep the nil check it would work (with an additional change in `util.go` to explicitly set an empty slice), but that will override all the default values because arguments that are not explicitly set seem to end up as empty strings instead of `nil`. Do you think it would be easy to adjust that non-set arguments would be `nil` instead of empty strings? I don't know the code base well enough yet and I have not so much experience with the `flag` package.

[the remark this answers is not in the document]
Comment (the) [reply runner/flag_test.go:105]: If I see correctly it's only possible to define string variable flags with a string as default value (not a pointer, thus impossible to set `nil` as default value).

See also here https://github.com/cosmtrek/air/blob/master/runner/flag.go#L12.

**Comment by the** [comment]
> @the use comma to split string as slice is a good idea! plz fix the lint, and add the test case which include empty string, and seems should add more description in README.
> 
> after that, I will merge this PR. thanks

I added a small note in the README.

File: README-zh_cn.md [hunk 2]
```diff
@@ -30,6 +30,10 @@ Air 是为 Go 应用开发设计的另外一个热重载的命令行工具。只
 
 `air --build.cmd "go build -o bin/api cmd/run.go" --build.bin "./bin/api"`
 
+对于以列表形式输入的参数，使用逗号来分隔项目:
```
Comment (xiantang) [inline README-zh_cn.md:33]: great!

File: runner/flag_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (xiantang) [reply runner/flag_test.go:105]: we can set default string as unset valuehttps://github.com/cosmtrek/air/pull/300/files#diff-19e19f16fa5018b7ad46fc960cdbe0138b0029dac020b42c798a9ff9b8203493R7 which is user use to set.

then filter the unset value in there https://github.com/cosmtrek/air/pull/300/files#diff-0a9b65ef9e6a4d3b9e2aec07d2dbaed8bb3866d74d3a2c78d78ddf56ed9aa071L324 

u can cherry-pick my commit

**Comment by xiantang** [comment]
https://github.com/cosmtrek/air/pull/300#

File: runner/flag_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (the) [reply runner/flag_test.go:105]: Thanks for your support. I cherry-picked all 3 commits.
